File: app/src/main/java/com/projects/andreafranco/imagegalleryapp/models/Image.java
package com.projects.andreafranco.imagegalleryapp.models;

import android.graphics.Bitmap;
import android.os.Parcel;
import android.os.Parcelable;

import java.util.Arrays;

public class Image implements Parcelable{

    private String mTitle;
    private String mSize;
    private int mWidth;
    private int mHeight;
    private Bitmap mImage;

    public Image(String title, String size, int width, int height, Bitmap image) {
        mTitle = title;
        mSize = size;
        mWidth = width;
        mHeight = height;
        mImage = image;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        this.mTitle = title;
    }

    public int getWidth() {
        return mWidth;
    }

    public void setHeight(int height) {
        this.mHeight = height;
    }

    public int getHeight() {
        return mHeight;
    }

    public void setWidth(int width) {
        this.mWidth = width;
    }

    public Bitmap getImage() {
        return mImage;
    }

    public void setmImage(Bitmap image) {
        this.mImage = image;
    }

    public String getSize() {
        return mSize;
    }

    public void setSize(String size) {
        this.mSize = size;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(this.mImage, flags);
        dest.writeString(this.mTitle);
        dest.writeString(this.mSize);
        dest.writeInt(this.mWidth);
        dest.writeInt(this.mHeight);
    }

    protected Image(Parcel in) {
        this.mImage = in.readParcelable(null);
        this.mTitle = in.readString();
        this.mSize = in.readString();
        this.mWidth= in.readInt();
        this.mHeight= in.readInt();
    }

    public static final Parcelable.Creator<Image> CREATOR = new Parcelable.Creator<Image>() {
        @Override
        public Image createFromParcel(Parcel source) {
            return new Image(source);
        }

        @Override
        public Image[] newArray(int size) {
            return new Image[size];
        }
    };
}

File: app/src/main/java/com/projects/andreafranco/imagegalleryapp/models/Image.kt
package com.projects.andreafranco.imagegalleryapp.models

import android.graphics.Bitmap
import android.os.Parcel
import android.os.Parcelable

class Image : Parcelable {

    var title: String? = null
    var size: Double = 0.toDouble()
    var width: Int = 0
    var height: Int = 0
    var image: Bitmap? = null
        private set

    constructor(title: String, size: Double, width: Int, height: Int, image: Bitmap) {
        this.title = title
        this.size = size
        this.width = width
        this.height = height
        this.image = image
    }

    fun setmImage(image: Bitmap) {
        this.image = image
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(this.image, flags)
        dest.writeString(this.title)
        dest.writeDouble(this.size)
        dest.writeInt(this.width)
        dest.writeInt(this.height)
    }

    protected constructor(`in`: Parcel) {
        this.image = `in`.readParcelable(null)
        this.title = `in`.readString()
        this.size = `in`.readDouble()
        this.width = `in`.readInt()
        this.height = `in`.readInt()
    }

    companion object CREATOR : Parcelable.Creator<Image> {
            override fun createFromParcel(source: Parcel): Image {
                return Image(source)
            }

            override fun newArray(size: Int): Array<Image?> {
                return arrayOfNulls(size)
            }
        }
}

